src/core/output-layout.cpp

```diff
@@ -1292,6 +1293,28 @@ class output_layout_t::impl
             remove_output(output);
         });
 
+        lo->on_request_state.set_callback([=] (void *data)
+        {
+            auto ev = static_cast<wlr_output_event_request_state*>(data);
+            wlr_output_commit_state(ev->output, ev->state);
+            send_wlr_configuration();
+            for (auto& output : outputs)
+            {
+                output.second->emit_configuration_changed(
+                    wf::OUTPUT_POSITION_CHANGE | wf::OUTPUT_MODE_CHANGE);
+            }
+
+            output_layout_configuration_changed_signal signal;
+            wf::get_core().output_layout->emit(&signal);
+            if (!lo->output)
```

Comment:
We should check this maybe after the output_commit_state, because the emit_configuration_changed functions also don't have a check for whether the output is NULL, we can either add a check there too, or move this check upwards in this function so that we don't emit_configuration_changed on an output with NULL wf::output_t object.